src/metadata.rs

```diff
@@ -76,6 +79,173 @@ fn path_to_content_type(path: &Path) -> String {
 }
 
 impl Metadata21 {
+    /// Merge metadata with pyproject.toml
+    ///
+    /// manifest_path must be the directory, not the file
+    fn merge_pyproject_toml(&mut self, manifest_path: impl AsRef<Path>) -> Result<()> {
+        let manifest_path = manifest_path.as_ref();
+        if let Ok(pyproject_toml) = PyProjectToml::new(manifest_path) {
```

Comment:
This should return an error if pyproject.toml is invalid and only ignore if there isn't a pyproject.toml, i.e. something like

```rust
if !manifest_path.joinpath("pyproject.toml").is_file() {
    return Ok(());
}
let pyproject_toml = PyProjectToml::new(manifest_path).context("pyproject.toml is invalid")?;
```